proposals/interface-types/working-notes/choices/webidl.md

```diff
@@ -0,0 +1,257 @@
+
+### Nominal Types
+
+Javascript does not have a static model of types. On the other hand, many
+programming languages do have static types. One of the features of many static type systems is the so-called _nominal type_.
+
+A nominal type differs from a structural type in several ways; but one of the
+most fundamental ways is that a nominal type can be used to model entities that
+are not directly conveyed by the actual data used to denote them.
+
+For example, one may choose to model a chair using the nominal type `Chair`;
+whose values contain a pair of integers: the SKU and price (for example). The
+pair of integers is _not equivalent_ to a chair, but represent the information
+needed to model chairs in some application or API. 
+
+A different entity (a person say) may also be modeled by a type `Person` which
+contains two integers also (for example, the age of the person and an index into
+a street address table).
+
+A given pair of integers might be either a `Chair` or a `Person` (or a 2D point)
+and there is no way of conveying the distinction unless the nominal type is also
+taken into consideration.
+
+### Algebraic Type Definitions
+
+The IT type language permits the definition of a limited form of _algebraic type
+definition_. This is, in part, to regularize concepts such as nullability and to
+permit limited forms of polymorphism (allowing a color to be named by a string,
+a vector of small integers or a large integer, for example).
+
+WebIDL explicitly prohibits the formation of new types; indeed it uses type
+unions to achieve similar polymorphism. However, as we note above, not all
+programming languages support dynamic types or type unions; which makes it
+problematic to support such polymorphism.
+
+The form of algebraic types envisaged in the IT system is intentionally
+limited. For example, we do not permit _recursive_ types to be defined. This is
+in keeping with the vision of supporting APIs across ownership boundaries where
+complex data is likely to take limited form.[^This design choice may be
+revisited in future versions of the IT specification if supporting structured
+types such as abstract XML and/or JSON become important.]
+
+### Working with WebIDL
+
+While the IT language introduces its own IDL, it is still clearly important to
+support WebIDL; particularly for Web APIs.
+
+The strategy that we adopt can be summarized as interpreting WebIDL as an
+alternate target to webAssembly itself.
+
+When using IT to specify a public API for a webAssembly module, the author must
+provide special _adapters_ -- import adapters for accessing external APIs and
+export adapters when offering a public API.
+
+Similarly, in the case where a WebIDL API is involved, there would be adapters
+between the IT language and WebIDL. There would also be adapters between WebIDL
+and the host environment; although these are usually private to the host.
+
+In many cases, the adapters would be automatically generated by language
```

Comment:
This paragraph is confusing; aren't the adapters from IT <-> WebIDL provided by a web VM, not language-specific tools?

Reply:
Agreed. This needs redoing a bit. The current ordering is result of a refactor that got mangled.